test/subdataframe_mutation.jl

```diff
@@ -0,0 +1,1647 @@
+end
+
+@testset "select! on GroupedDataFrame{SubDataFrame}" begin
+    df = DataFrame(a=[1, 1, 1, 2, 2, 3], b=11:16, c=21:26, d=31:36)
+    sdf = @view df[2:4, [2, 1, 3]]
+    gsdf = groupby(sdf, :a)
+    @test_throws ArgumentError select!(gsdf, :b => x -> x .+ 100, renamecols=false)
+    @test df == DataFrame(a=[1, 1, 1, 2, 2, 3], b=11:16, c=21:26, d=31:36)
+    sdf = @view df[2:4, [2, 1]]
+    gsdf = groupby(sdf, :a)
+    @test_throws ArgumentError select!(gsdf, :b => x -> x .+ 100, renamecols=false)
+    @test df == DataFrame(a=[1, 1, 1, 2, 2, 3], b=11:16, c=21:26, d=31:36)
+    sdf = @view df[2:4, [1, 3, 2]]
+    gsdf = groupby(sdf, :a)
+    @test select!(gsdf, :c, :b => x -> x .+ 100, renamecols=false, ungroup=false) === gsdf
+    @test df == DataFrame(a=[1, 1, 1, 2, 2, 3],
+                          b=[11, 112, 113, 114, 15, 16], c=21:26, d=31:36)
+
+    df = DataFrame(a=[1, 1, 1, 2, 2, 3], b=11:16, c=21:26, d=31:36)
+    sdf = @view df[[3, 4, 2], :]
+    gsdf = groupby(sdf, :a)
+    @test select!(gsdf, :b => x -> x .+ 100, :c => :e, renamecols=false) === sdf
+    @test df ≅ DataFrame(a=[1, 1, 1, 2, 2, 3],
+                         b=[11, 112, 113, 114, 15, 16],
+                         e=[missing, 22, 23, 24, missing, missing])
+end
+
+@testset "transform! on GroupedDataFrame{SubDataFrame}" begin
+    df = DataFrame(a=[1, 1, 1, 2, 2, 3], b=11:16, c=21:26, d=31:36)
+    sdf = @view df[2:4, [2, 1, 3]]
+    gsdf = groupby(sdf, :a)
+    transform!(gsdf, :b => x -> x .+ 100, renamecols=false)
+    @test df == DataFrame(a=[1, 1, 1, 2, 2, 3],
+                          b=[11, 112, 113, 114, 15, 16], c=21:26, d=31:36)
+
+    df = DataFrame(a=[1, 1, 1, 2, 2, 3], b=11:16, c=21:26, d=31:36)
+    sdf = @view df[2:4, [2, 1]]
+    gsdf = groupby(sdf, :a)
+    transform!(gsdf, :b => x -> x .+ 100, renamecols=false)
+    @test df == DataFrame(a=[1, 1, 1, 2, 2, 3],
+                          b=[11, 112, 113, 114, 15, 16], c=21:26, d=31:36)
+
+    df = DataFrame(a=[1, 1, 1, 2, 2, 3], b=11:16, c=21:26, d=31:36)
+    sdf = @view df[2:4, [1, 3, 2]]
+    gsdf = groupby(sdf, :a)
+    @test transform!(gsdf, :c, :b => x -> x .+ 100, renamecols=false, ungroup=false) == gsdf
+    @test df == DataFrame(a=[1, 1, 1, 2, 2, 3],
+                          b=[11, 112, 113, 114, 15, 16], c=21:26, d=31:36)
+
+    df = DataFrame(a=[1, 1, 1, 2, 2, 3], b=11:16, c=21:26, d=31:36)
+    sdf = @view df[[3, 4, 2], :]
+    gsdf = groupby(sdf, :a)
+    @test transform!(gsdf, :b => x -> x .+ 100, :c => :e, renamecols=false) === sdf
+    @test df ≅ DataFrame(a=[1, 1, 1, 2, 2, 3],
+                         b=[11, 112, 113, 114, 15, 16],
+                         c=21:26, d=31:36,
+                         e=[missing, 22, 23, 24, missing, missing])
+end
+
+@testset "promote_type tests" begin
+    df = DataFrame(a=1:4)
+    sdf = @view df[1:1, :]
+    sdf[!, 1] = [1.5]
+    @test df.a == [1.5, 2, 3, 4]
+    @test eltype(df.a) === Float64
+
+    # note that CategoricalVector is dropped as
```

Comment:
@nalimilan - this part of tests probably deserves your special attention. Thank you!